extensions-contrib/opentelemetry-emitter/src/main/java/org/apache/druid/emitter/opentelemetry/OpenTelemetryEmitter.java

```diff
@@ -0,0 +1,126 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.emitter.opentelemetry;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import org.apache.druid.java.util.common.logger.Logger;
+import org.apache.druid.java.util.emitter.core.Emitter;
+import org.apache.druid.java.util.emitter.core.Event;
+import org.apache.druid.java.util.emitter.service.ServiceMetricEvent;
+import org.joda.time.DateTime;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class OpenTelemetryEmitter implements Emitter
+{
+  private static final Logger log = new Logger(OpenTelemetryEmitter.class);
+  private final Tracer tracer;
+  private final TextMapPropagator propagator;
+
+  OpenTelemetryEmitter(OpenTelemetry openTelemetry)
+  {
+    tracer = openTelemetry.getTracer("druid-opentelemetry-extension");
+    propagator = openTelemetry.getPropagators().getTextMapPropagator();
+  }
+
+  @Override
+  public void start()
+  {
+    log.debug("Starting OpenTelemetryEmitter");
+  }
+
+  @Override
+  public void emit(Event e)
+  {
+    if (!(e instanceof ServiceMetricEvent)) {
+      return;
+    }
+    ServiceMetricEvent event = (ServiceMetricEvent) e;
+
+    // We generate spans for the following types of events:
+    // query/time
+    if (!event.getMetric().equals("query/time")) {
+      return;
+    }
+
+    emitQueryTimeEvent(event);
+  }
+
+  private void emitQueryTimeEvent(ServiceMetricEvent event)
+  {
+    DruidContextTextMapGetter druidContextTextMapGetter = new DruidContextTextMapGetter();
+    Context opentelemetryContext = propagator.extract(Context.current(), event, druidContextTextMapGetter);
+
+    try (Scope scope = opentelemetryContext.makeCurrent()) {
+      DateTime endTime = event.getCreatedTime();
+      DateTime startTime = endTime.minusMillis(event.getValue().intValue());
+
+      Span span = tracer.spanBuilder(event.getService())
+                        .setStartTimestamp(startTime.getMillis(), TimeUnit.MILLISECONDS)
+                        .startSpan();
+      Set<String> extractedKeys = druidContextTextMapGetter.getExtractedKeys();
+
+      getContextAsString(event).entrySet()
+                               .stream()
+                               .filter(entry -> !extractedKeys.contains(entry.getKey()))
+                               .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue))
+                               .forEach(span::setAttribute);
```

Comment:
I would in line this since `getContextAsString()` is only used here.  Also the method name is misleading since it gets the context as a `Map<String, String>` not as a single String.

```suggestion
      getContext(event).entrySet()
                       .stream()
                       .filter(entry -> entry.getValue() != null)
                       .filter(entry -> !extractedKeys.contains(entry.getKey()))
                       .forEach(entry -> span.setAttribute(entry.getKey(), entry.getValue().toString());
```